### PR TITLE
refactor(ffi): Simplify FFI usage

### PR DESCRIPTION
**Pull request by lukapeschke**
Simplification of the changes introduced by #74 :slightly_smiling_face: 

Memory profile still looks good
Figure_1

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -17,7 +17,7 @@ pyo3 = { version = "0.18.1", features = ["extension-module", "anyhow"] }
 version = "33.0.0"
 # There's a lot of stuff we don't want here, such as serde support
 default-features = false
-features = ["ipc", "ffi"]
+features = ["ipc", "ffi", "pyarrow"]
```
Comment (jGundermann) [inline Cargo.toml:20]: Maybe we dont even need ffi there ?

**Review by jGundermann — COMMENTED** [review]
Even better! thx for this.

File: Cargo.toml (hunk 1 above)
Comment (lukapeschke) [reply Cargo.toml]: you're right, I also removed IPC :slightly_smiling_face:

**Review by jGundermann — APPROVED** [review]
LGTM

**Pull request merged**